services/QuillLMS/app/models/concerns/teacher.rb

```diff
@@ -457,9 +457,9 @@ def teachers_activity_sessions_since_trial_start_date
   end
 
   def trial_days_remaining
-    valid_subscription =   subscription && subscription.expiration > Date.today
+    valid_subscription =   subscription && subscription.expiration > Date.current
```

Comment:
So there are a bunch of edge cases in this PR, that might be nice to fix here of the form: If you use `Date.current` or `Time.current` more than once in a method, you should assign it to a local var first, otherwise there is a rare edge case in which time passes between the calls and they could be different.

e.g.
```ruby
today = Date.current
# use today throughout method.

Reply:
Good idea.  I will take a look